pkg/registries/registries.go

```diff
@@ -155,10 +203,31 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// Set scope to Prefix if it exists
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if scopeIsNestedInsideScope(scope, blockedReg) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// IsValidRegistryOrWildcardScope returns true if scope is a valid registry or wildcard entry such as "*.registry.com"
+// This function can be used to validate the registries entries prior to calling EditRegistriesConfig
+// in the MCO or builds code
+func IsValidRegistryOrWildcardScope(scope string) bool {
```

Comment:
After https://issues.redhat.com/browse/OCPNODE-461 , doesn’t this become a generalized `IsValidRegistriesConfScope` that should accept wildcards, host[:port], as well as host[:port]/namespace[/…][/repo] ?

Reply:
Fixed